### ISSUE TITLE
Skip already installed libraries when validating uploaded packages

### ISSUE THREAD

**Issue by sr258** [Lumieducation/H5P-Nodejs-library#540]
To speed up validation, we shoudl skip already installed libraries that don't contain any updates.

**Comment by sr258** [comment]
Some profiling of the PackageImporter revealed that validation isn't the only cause of the slow upload time. This is how long things take in processProcess (over several runs with different packages) in ms:

```
validated - 88
extracted package - 105
read dir - 0
installed libraries - 2
finally - 19
validated - 83
extracted package - 140
read dir - 0
installed libraries - 4
finally - 21
validated - 97
extracted package - 153
read dir - 0
installed libraries - 7
finally - 7
validated - 38
extracted package - 63
read dir - 0
installed libraries - 1
finally - 3
validated - 36
extracted package - 67
read dir - 0
installed libraries - 2
finally - 1
validated - 84
extracted package - 158
read dir - 0
installed libraries - 4
finally - 13
validated - 950
extracted package - 1532
read dir - 0
installed libraries - 34
finally - 61
validated - 54
extracted package - 79
read dir - 0
installed libraries - 2
finally - 2
validated - 61
extracted package - 111
read dir - 0
installed libraries - 2
finally - 7
validated - 1170
extracted package - 1988
read dir - 0
installed libraries - 47
finally - 78
validated - 1251
extracted package - 2362
read dir - 0
installed libraries - 91
finally - 46
validated - 236
extracted package - 206
read dir - 0
installed libraries - 11
finally - 46
validated - 80
extracted package - 164
read dir - 0
installed libraries - 5
finally - 11
validated - 128
extracted package - 322
read dir - 0
installed libraries - 17
finally - 12
validated - 92
extracted package - 257
read dir - 0
installed libraries - 5
finally - 4
validated - 40
extracted package - 108
read dir - 0
installed libraries - 3
finally - 15
validated - 38
extracted package - 96
read dir - 0
installed libraries - 5
finally - 16
validated - 21
extracted package - 60
read dir - 0
installed libraries - 2
finally - 14
validated - 47
extracted package - 97
read dir - 0
installed libraries - 7
finally - 28
validated - 40
extracted package - 64
read dir - 0
installed libraries - 1
finally - 3
validated - 52
extracted package - 179
read dir - 0
installed libraries - 4
finally - 3
validated - 69
extracted package - 174
read dir - 0
installed libraries - 8
finally - 12
validated - 55
extracted package - 106
read dir - 0
installed libraries - 3
finally - 17
validated - 33
extracted package - 80
read dir - 0
installed libraries - 4
finally - 2
validated - 113
extracted package - 136
read dir - 0
installed libraries - 2
finally - 40
validated - 48
extracted package - 96
read dir - 0
installed libraries - 4
finally - 18
validated - 46
extracted package - 81
read dir - 0
installed libraries - 2
finally - 8
validated - 436
extracted package - 647
read dir - 0
installed libraries - 16
finally - 61
validated - 1019
extracted package - 1801
read dir - 0
installed libraries - 44
finally - 47
```

Validation can take some time, particularly in content types with lots of library dependencies (Course Presentation, Interactive Book, ...).

It looks like the caching solution for library metadata helps a lot already, so it's probably not worth optimizing here.

**Comment by sr258** [comment]
Closed by  #1427
